vars/testsPublishResults.groovy

```diff
@@ -52,32 +52,29 @@ import groovy.transform.Field
 @GenerateDocumentation
 void call(Map parameters = [:]) {
     handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
-        def script = checkScript(this, parameters) ?: this
 
         prepare(parameters)
 
         // load default & individual configuration
         Map configuration = ConfigurationHelper.newInstance(this)
             .loadStepDefaults()
-            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
-            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
-            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName ?: env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixinGeneralConfig(GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(STEP_CONFIG_KEYS)
+            .mixinStageConfig(parameters.stageName ?: env.STAGE_NAME, STEP_CONFIG_KEYS)
             .mixin(parameters, PARAMETER_KEYS)
             .use()
 
         new Utils().pushToSWA([
             step: STEP_NAME,
-            stepParamKey1: 'scriptMissing',
-            stepParam1: parameters?.script == null
         ], configuration)
 
         publishJUnitReport(configuration.get('junit'))
         publishJacocoReport(configuration.get('jacoco'))
         publishCoberturaReport(configuration.get('cobertura'))
         publishJMeterReport(configuration.get('jmeter'))
 
-        if (configuration.failOnError && JenkinsUtils.hasTestFailures(script.currentBuild)) {
-            script.currentBuild.result = 'FAILURE'
+        if (configuration.failOnError && JenkinsUtils.hasTestFailures(currentBuild)) {
+            currentBuild.result = 'FAILURE'
```

Comment:
are we sure that this is set on the correct instance of `currentBuild`?

Reply:
yes. See [here](https://github.com/jenkinsci/workflow-support-plugin/blob/261e3109e4d160f60da399374da9e9621f4d141e/src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java#L96). Status is not manipulated on the instance of `RunWrapper`, but status is updated on `hudson.model.Run`. In fact we have several instances of RunWrapper, but only one common instance for `Run` when running one build.